src/main/java/io/github/genomicdatainfrastructure/discovery/datasets/infra/ckan/persistence/CkanDatasetIdsCollector.java

```diff
@@ -2,46 +2,53 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
-package io.github.genomicdatainfrastructure.discovery.repositories;
+package io.github.genomicdatainfrastructure.discovery.datasets.infra.ckan.persistence;
 
+import io.github.genomicdatainfrastructure.discovery.datasets.applications.ports.DatasetIdsCollector;
 import io.github.genomicdatainfrastructure.discovery.model.DatasetSearchQuery;
-import io.github.genomicdatainfrastructure.discovery.model.DatasetsSearchResponse;
 import io.github.genomicdatainfrastructure.discovery.remote.ckan.api.CkanQueryApi;
+import io.github.genomicdatainfrastructure.discovery.remote.ckan.model.CkanPackage;
 import io.github.genomicdatainfrastructure.discovery.utils.CkanFacetsQueryBuilder;
-import io.github.genomicdatainfrastructure.discovery.utils.PackagesSearchResponseMapper;
-import io.quarkus.arc.lookup.LookupIfProperty;
 import jakarta.enterprise.context.ApplicationScoped;
 import jakarta.inject.Inject;
 import org.eclipse.microprofile.rest.client.inject.RestClient;
 
-@LookupIfProperty(name = "sources.ckan", stringValue = "true")
+import java.util.List;
+
 @ApplicationScoped
-public class CkanDatasetsRepository implements DatasetsRepository {
+public class CkanDatasetIdsCollector implements DatasetIdsCollector {
 
-    private static final String SELECTED_FACETS = "[\"access_rights\",\"theme\",\"tags\",\"spatial_uri\",\"organization\",\"publisher_name\",\"res_format\"]";
     private final CkanQueryApi ckanQueryApi;
 
     @Inject
-    public CkanDatasetsRepository(
+    public CkanDatasetIdsCollector(
             @RestClient CkanQueryApi ckanQueryApi
     ) {
         this.ckanQueryApi = ckanQueryApi;
     }
 
     @Override
-    public DatasetsSearchResponse search(DatasetSearchQuery query, String accessToken) {
+    public List<String> collect(DatasetSearchQuery query, String accessToken) {
         var facetsQuery = CkanFacetsQueryBuilder.buildFacetQuery(query);
 
         var response = ckanQueryApi.packageSearch(
                 query.getQuery(),
                 facetsQuery,
-                query.getSort(),
-                query.getRows(),
-                query.getStart(),
-                SELECTED_FACETS,
+                null,
+                null,
```

Comment:
question: should rows be null here? By default, package search uses 10 rows.

Reply:
suggestion: you should check CKAN API to remove this limit. You could also check if it is possible to reduce the fields we retrieve. In t his context, we only need `identifier`.